### PR TITLE
Spawn hints in neighborhood of open fields

### PR DESCRIPTION
**Pull request by gboehl**
This is a proposal to address issue #305. As stated there, hints appear at any random field that is not open and has no marker. Hints out in the void which are not in the neighborhood of any open field are never useful. I thus propose to only show hints that, if possible, are adjacent to a disclosed field.

Thanks for creating this nice app!

### CODE REVIEW

**Comment by lucasnlm** [comment]
Nice! Thanks for your contribution!

File: common/src/main/java/dev/lucasnlm/antimine/common/level/logic/MinefieldHandler.kt [hunk 1]
```diff
@@ -40,14 +40,39 @@ class MinefieldHandler(
...
 
+    fun hasUncoveredNeighbor( index: Int ): Boolean {
+        val numOfUncoveredNeighbors = field.getOrNull(index)?.run {
+            neighborsIds
+                .map { field[it] }
+                .filter { !it.isCovered }
+                .count() 
+        } 
+        return numOfUncoveredNeighbors!! > 0
+    }
```
Comment (lucasnlm) [inline common/src/main/java/dev/lucasnlm/antimine/common/level/logic/MinefieldHandler.kt:51]: I think you can do this, instead:

```kt
    private fun hasUncoveredNeighbor(area: Area): Boolean {
        return area.neighborsIds.map { field[it] }.count { !it.isCovered } > 0
    }
```

File: common/src/main/java/dev/lucasnlm/antimine/common/level/logic/MinefieldHandler.kt [hunk 2]
```diff
@@ -40,14 +40,39 @@ class MinefieldHandler(
...
+        }
+        val unrevealedMinesWithUncoveredNeighbors = unrevealedMines.filter { 
+            hasUncoveredNeighbor(it.id)
+        }
+        val potentialTargets = 
+            if (unrevealedMinesWithUncoveredNeighbors.count() > 0) { 
+                unrevealedMinesWithUncoveredNeighbors 
+            } else {
+                unrevealedMines
+            }
```
Comment (lucasnlm) [inline common/src/main/java/dev/lucasnlm/antimine/common/level/logic/MinefieldHandler.kt:71]: With that suggestion, you can call here:
```kt
        val unrevealedMinesWithUncoveredNeighbors = unrevealedMines.filter(::hasUncoveredNeighbor)
        val potentialTargets = unrevealedMinesWithUncoveredNeighbors.ifEmpty {
            unrevealedMines
        }
```

File: common/src/main/java/dev/lucasnlm/antimine/common/level/logic/MinefieldHandler.kt [hunk 3]
```diff
@@ -40,14 +40,39 @@ class MinefieldHandler(
...
+
         val nearestTarget =
             if (lastX != null && lastY != null) {
-                unrevealedMines.filter {
+                potentialTargets.filter {
                     (lastX - it.posX).absoluteValue < NEAR_MINE_THRESHOLD &&
                         (lastY - it.posY).absoluteValue < NEAR_MINE_THRESHOLD
```
Comment (lucasnlm) [inline common/src/main/java/dev/lucasnlm/antimine/common/level/logic/MinefieldHandler.kt:77]: In the future, I would like to prioritize the mines that are visible in the screen.

**Comment by lucasnlm** [comment]
@gboehl Your suggestion looks functional. Please, let me know if you will implement the suggestions. I can finish for you if you are ok.
Don't bother with `Check Code Quality` BTW.

**Comment by gboehl** [comment]
Thanks for the suggestions! I actually have zero previous experience with kotlin, so this is much appreciated. I added your suggestions and the PR should now be good to go.

**Comment by gboehl** [comment]
This should close #305

**Pull request merged**